### PR TITLE
Synapse spark tests

### PR DESCRIPTION
**Pull request by jianghaolu**

### CODE REVIEW

File: sdk/synapse/azure-analytics-synapse-spark/src/test/java/com/azure/analytics/synapse/spark/SparkSessionAsyncClientTest.java [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                .verifyComplete();
+
+            // arrange
+            String code = "print('hello, Azure CLI')";
+
+            SparkStatementOptions options = new SparkStatementOptions()
+                .setKind(SparkStatementLanguageType.PYSPARK)
+                .setCode(code);
+
+            Mono<SparkStatement> createSparkStatement = Mono.defer(() -> {
```
Comment (chamons) [inline sdk/synapse/azure-analytics-synapse-spark/src/test/java/com/azure/analytics/synapse/spark/SparkSessionAsyncClientTest.java:117]: I'm assuming this is the code you referred to as needing to be LRO'ed?

**Review by chamons — COMMENTED** [review]
Seems reasonable enough outside of the need to LRO the spark session/statement code at some point. 

Let me know if you need pointed to my C# version of either.

File: sdk/synapse/azure-analytics-synapse-spark/src/test/java/com/azure/analytics/synapse/spark/SparkSessionAsyncClientTest.java (hunk 1 above)
Comment (jianghaolu) [reply sdk/synapse/azure-analytics-synapse-spark/src/test/java/com/azure/analytics/synapse/spark/SparkSessionAsyncClientTest.java:117]: This I'm not sure. I did exactly the same as the tests in CLI to wait for 6 minutes. The operations themselves are not marked with `x-ms-long-running-operation`: https://github.com/Azure/azure-rest-api-specs/blob/master/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L462. They might be - but I'm not too sure until I understand more of the [standard LRO patterns](https://github.com/Microsoft/api-guidelines/blob/master/Guidelines.md#13-long-running-operations).

**Comment by JonathanGiles** [comment]
@jianghaolu What's the status here?

**Comment by jianghaolu** [comment]
These tests are already in `main`.

**Pull request closed without merging**